From 4fa52b3ceff6268dca5bd49db2c8b571d6bd0332 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 23:17:30 -0400
Subject: [scsi] cxgb4i: remove spurious use of rcu

Message-id: <1410477451-95647-6-git-send-email-svemuri@redhat.com>
Patchwork-id: 93910
O-Subject: [RHEL7.1 PATCH BZ 1126539 5/6] cxgb4i : remove spurious use of rcu
Bugzilla: 1126539
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

As pointed out by the intel guys, there is no need to hold rcu read lock in
cxgbi_inet6addr_handler(), this patch removes it.

Fixes: 759a0cc5a3e1 ("cxgb4i: Add ipv6 code to driver, call into libcxgbi ipv6 api")
Signed-off-by: Anish Bhatt <anish@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e6b92c25d20c64c271ef429bba8febeefb848b5b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
index 7a30597..a4e4e2a 100644
--- a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
+++ b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
@@ -1644,25 +1644,22 @@ static int cxgbi_inet6addr_handler(struct notifier_block *this,
  struct cxgbi_device *cdev;
  int ret = NOTIFY_DONE;
 
- rcu_read_lock();
-
  if (event_dev->priv_flags & IFF_802_1Q_VLAN)
   event_dev = vlan_dev_real_dev(event_dev);
 
  cdev = cxgbi_device_find_by_netdev(event_dev, NULL);
- if (!cdev) {
-  rcu_read_unlock();
+
+ if (!cdev)
   return ret;
- }
+
  switch (event) {
  case NETDEV_UP:
   ret = cxgb4_clip_get(event_dev,
          (const struct in6_addr *)
          ((ifa)->addr.s6_addr));
-  if (ret < 0) {
-   rcu_read_unlock();
+  if (ret < 0)
    return ret;
-  }
+
   ret = NOTIFY_OK;
   break;
 
@@ -1677,7 +1674,6 @@ static int cxgbi_inet6addr_handler(struct notifier_block *this,
   break;
  }
 
- rcu_read_unlock();
  return ret;
 }
 
-- 
1.7.1